Demo Polyhedron failed to build

Hi, Tried to build CGAL 4.10 from source against qt-5.9.0  (and qt-5.9.1)  Make demos failed with
```
/home/bee/Downloads/CGAL-4.10/demo/Polyhedron/MainWindow.cpp: In member function ‘virtual void MainWindow::open(QString)’:
/home/bee/Downloads/CGAL-4.10/demo/Polyhedron/MainWindow.cpp:1027:34: error: no matching function for call to ‘QOpenGLContext::makeCurrent()’
   viewer->context()->makeCurrent();
                                  ^
In file included from /home/bee/opt/qt5/include/QtGui/qopenglfunctions_2_1.h:58:0,
                 from /home/bee/opt/qt5/include/QtGui/QOpenGLFunctions_2_1:1,
                 from /home/bee/Downloads/CGAL-4.10/demo/Polyhedron/../../include/CGAL/Three/Viewer_interface.h:30,
                 from /home/bee/Downloads/CGAL-4.10/demo/Polyhedron/../../include/CGAL/Three/TextRenderer.h:30,
                 from /home/bee/Downloads/CGAL-4.10/demo/Polyhedron/MainWindow.cpp:7:
/home/bee/opt/qt5/include/QtGui/qopenglcontext.h:172:10: note: candidate: bool QOpenGLContext::makeCurrent(QSurface*)
     bool makeCurrent(QSurface *surface);
          ^
/home/bee/opt/qt5/include/QtGui/qopenglcontext.h:172:10: note:   candidate expects 1 argument, 0 provided
demo/Polyhedron/CMakeFiles/polyhedron_demo.dir/build.make:134: recipe for target 'demo/Polyhedron/CMakeFiles/polyhedron_demo.dir/MainWindow.cpp.o' failed
make[3]: *** [demo/Polyhedron/CMakeFiles/polyhedron_demo.dir/MainWindow.cpp.o] Error 1
CMakeFiles/Makefile2:28944: recipe for target 'demo/Polyhedron/CMakeFiles/polyhedron_demo.dir/all' failed
make[2]: *** [demo/Polyhedron/CMakeFiles/polyhedron_demo.dir/all] Error 2
CMakeFiles/Makefile2:960: recipe for target 'CMakeFiles/demos.dir/rule' failed
make[1]: *** [CMakeFiles/demos.dir/rule] Error 2
Makefile:188: recipe for target 'demos' failed
make: *** [demos] Error 2
``` 

CGAL builds without problem.
OS is Ubuntu 16.04 64 bits.

--- a/AABB_tree/demo/AABB_tree/MainWindow.cpp
+++ b/AABB_tree/demo/AABB_tree/MainWindow.cpp
@@ -424,7 +424,12 @@ void MainWindow::on_actionCopy_snapshot_triggered()
   QClipboard *qb = QApplication::clipboard();
   m_pViewer->makeCurrent();
   m_pViewer->raise();
+#if QGLVIEWER_VERSION >= 0x020700
+  QImage snapshot = m_pViewer->grabFramebuffer();
+#else
   QImage snapshot = m_pViewer->grabFrameBuffer(true);
+
+#endif
   qb->setImage(snapshot);
 	QApplication::restoreOverrideCursor();
 }

--- a/AABB_tree/demo/AABB_tree/Scene.h
+++ b/AABB_tree/demo/AABB_tree/Scene.h
@@ -16,6 +16,7 @@
 #include <CGAL/AABB_face_graph_triangle_primitive.h>
 
 #include <QtCore/qglobal.h>
+#include <QMap>
 #include <QGLViewer/manipulatedFrame.h>
 #include <QGLViewer/qglviewer.h>
 #include <QOpenGLFunctions_2_1>

--- a/AABB_tree/demo/AABB_tree/Viewer.h
+++ b/AABB_tree/demo/AABB_tree/Viewer.h
@@ -1,6 +1,6 @@
 #ifndef VIEWER_H
 #define VIEWER_H
-
+#include <QMap>
 #include <QGLViewer/qglviewer.h>
 
 

--- a/Alpha_shapes_3/demo/Alpha_shapes_3/Viewer.h
+++ b/Alpha_shapes_3/demo/Alpha_shapes_3/Viewer.h
@@ -2,6 +2,7 @@
 #define VIEWER_H
 
 #include "typedefs.h"
+#include <QMap>
 #include <QGLViewer/qglviewer.h>
 #include <QOpenGLFunctions_2_1>
 #include <QOpenGLVertexArrayObject>

--- a/Circular_kernel_3/demo/Circular_kernel_3/Viewer.h
+++ b/Circular_kernel_3/demo/Circular_kernel_3/Viewer.h
@@ -1,3 +1,4 @@
+#include <QMap>
 #include <QGLViewer/qglviewer.h>
 #include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
 

--- a/Linear_cell_complex/demo/Linear_cell_complex/Viewer.cpp
+++ b/Linear_cell_complex/demo/Linear_cell_complex/Viewer.cpp
@@ -663,7 +663,11 @@ void Viewer::sceneChanged()
   if (m_previous_scene_empty)
     this->showEntireScene();
   else
+#if QGLVIEWER_VERSION >= 0x020700
+    this->update();
+#else
     this->updateGL();
+#endif
 
   m_previous_scene_empty = scene->lcc->is_empty(); // for the next call to sceneChanged
 }
@@ -783,7 +787,11 @@ void Viewer::keyPressEvent(QKeyEvent *e)
       glPolygonMode(GL_FRONT_AND_BACK, GL_FILL);
       displayMessage("Filled faces.");
     }
+#if QGLVIEWER_VERSION >= 0x020700
+    update();
+#else
     updateGL();
+#endif
   }
   else if ((e->key()==Qt::Key_F) && (modifiers==Qt::NoButton))
   {
@@ -792,19 +800,31 @@ void Viewer::keyPressEvent(QKeyEvent *e)
       displayMessage("Flat shading.");
     else
       displayMessage("Gouraud shading.");
+#if QGLVIEWER_VERSION >= 0x020700
+    update();
+#else
     updateGL();
+#endif
   }
   else if ((e->key()==Qt::Key_E) && (modifiers==Qt::NoButton))
   {
     edges = !edges;
     displayMessage(QString("Draw edges=%1.").arg(edges?"true":"false"));
+#if QGLVIEWER_VERSION >= 0x020700
+    update();
+#else
     updateGL();
+#endif
   }
   else if ((e->key()==Qt::Key_V) && (modifiers==Qt::NoButton))
   {
     vertices = !vertices;
     displayMessage(QString("Draw vertices=%1.").arg(vertices?"true":"false"));
+#if QGLVIEWER_VERSION >= 0x020700
+    update();
+#else
     updateGL();
+#endif
   }
   else if ((e->key()==Qt::Key_N) && (modifiers==Qt::NoButton))
   {
@@ -816,25 +836,41 @@ void Viewer::keyPressEvent(QKeyEvent *e)
   {
     size_edges+=.5;
     displayMessage(QString("Size of edges=%1.").arg(size_edges));
+#if QGLVIEWER_VERSION >= 0x020700
+    update();
+#else
     updateGL();
+#endif
   }
   else if ((e->key()==Qt::Key_Minus) && (modifiers==Qt::KeypadModifier))
   {
     if (size_edges>.5) size_edges-=.5;
     displayMessage(QString("Size of edges=%1.").arg(size_edges));
+#if QGLVIEWER_VERSION >= 0x020700
+    update();
+#else
     updateGL();
+#endif
   }
   else if ((e->key()==Qt::Key_Plus) && (modifiers==(Qt::ShiftModifier|Qt::KeypadModifier)))
   {
     size_points+=.5;
     displayMessage(QString("Size of points=%1.").arg(size_points));
+#if QGLVIEWER_VERSION >= 0x020700
+    update();
+#else
     updateGL();
+#endif
   }
   else if ((e->key()==Qt::Key_Minus) && (modifiers==(Qt::ShiftModifier|Qt::KeypadModifier)))
   {
     if (size_points>.5) size_points-=.5;
     displayMessage(QString("Size of points=%1.").arg(size_points));
+#if QGLVIEWER_VERSION >= 0x020700
+    update();
+#else
     updateGL();
+#endif
   }
   else if ((e->key()==Qt::Key_PageUp) && (modifiers==Qt::NoButton))
   {
@@ -846,7 +882,11 @@ void Viewer::keyPressEvent(QKeyEvent *e)
     if (ambient.z()>1.) ambient.setZ(1.);
     displayMessage(QString("Light color=(%1 %2 %3).").
         arg(ambient.x()).arg(ambient.y()).arg(ambient.z()));
+#if QGLVIEWER_VERSION >= 0x020700
+    update();
+#else
     updateGL();
+#endif
   }
   else if ((e->key()==Qt::Key_PageDown) && (modifiers==Qt::NoButton))
   {
@@ -858,7 +898,11 @@ void Viewer::keyPressEvent(QKeyEvent *e)
     if (ambient.z()<0.) ambient.setZ(0.);
     displayMessage(QString("Light color=(%1 %2 %3).").
         arg(ambient.x()).arg(ambient.y()).arg(ambient.z()));
+#if QGLVIEWER_VERSION >= 0x020700
+    update();
+#else
     updateGL();
+#endif
   }
   else if ((e->key()==Qt::Key_PageUp) && (modifiers==Qt::ShiftModifier))
   {
@@ -866,7 +910,11 @@ void Viewer::keyPressEvent(QKeyEvent *e)
     if (ambient.x()>1.) ambient.setX(1.);
     displayMessage(QString("Light color=(%1 %2 %3).").
         arg(ambient.x()).arg(ambient.y()).arg(ambient.z()));
+#if QGLVIEWER_VERSION >= 0x020700
+    update();
+#else
     updateGL();
+#endif
   }
   else if ((e->key()==Qt::Key_PageUp) && (modifiers==Qt::AltModifier))
   {
@@ -874,7 +922,11 @@ void Viewer::keyPressEvent(QKeyEvent *e)
     if (ambient.y()>1.) ambient.setY(1.);
     displayMessage(QString("Light color=(%1 %2 %3).").
         arg(ambient.x()).arg(ambient.y()).arg(ambient.z()));
+#if QGLVIEWER_VERSION >= 0x020700
+    update();
+#else
     updateGL();
+#endif
   }
   else if ((e->key()==Qt::Key_PageUp) && (modifiers==Qt::ControlModifier))
   {
@@ -882,7 +934,11 @@ void Viewer::keyPressEvent(QKeyEvent *e)
     if (ambient.z()>1.) ambient.setZ(1.);
     displayMessage(QString("Light color=(%1 %2 %3).").
         arg(ambient.x()).arg(ambient.y()).arg(ambient.z()));
+#if QGLVIEWER_VERSION >= 0x020700
+    update();
+#else
     updateGL();
+#endif
   }
   else if ((e->key()==Qt::Key_PageDown) && (modifiers==Qt::ShiftModifier))
   {
@@ -890,7 +946,11 @@ void Viewer::keyPressEvent(QKeyEvent *e)
     if (ambient.x()<0.) ambient.setX(0.);
     displayMessage(QString("Light color=(%1 %2 %3).").
         arg(ambient.x()).arg(ambient.y()).arg(ambient.z()));
+#if QGLVIEWER_VERSION >= 0x020700
+    update();
+#else
     updateGL();
+#endif
   }
   else if ((e->key()==Qt::Key_PageDown) && (modifiers==Qt::AltModifier))
   {
@@ -898,7 +958,11 @@ void Viewer::keyPressEvent(QKeyEvent *e)
     if (ambient.y()<0.) ambient.setY(0.);
     displayMessage(QString("Light color=(%1 %2 %3).").
         arg(ambient.x()).arg(ambient.y()).arg(ambient.z()));
+#if QGLVIEWER_VERSION >= 0x020700
+    update();
+#else
     updateGL();
+#endif
   }
   else if ((e->key()==Qt::Key_PageDown) && (modifiers==Qt::ControlModifier))
   {
@@ -906,7 +970,11 @@ void Viewer::keyPressEvent(QKeyEvent *e)
     if (ambient.z()<0.) ambient.setZ(0.);
     displayMessage(QString("Light color=(%1 %2 %3).").
         arg(ambient.x()).arg(ambient.y()).arg(ambient.z()));
+#if QGLVIEWER_VERSION >= 0x020700
+    update();
+#else
     updateGL();
+#endif
   }
   else
     QGLViewer::keyPressEvent(e);

--- a/Linear_cell_complex/demo/Linear_cell_complex/Viewer.h
+++ b/Linear_cell_complex/demo/Linear_cell_complex/Viewer.h
@@ -24,6 +24,7 @@
 #include "typedefs.h"
 
 #include <vector>
+#include <QMap>
 #include <QGLViewer/qglviewer.h>
 #include <QKeyEvent>
 #include <QOpenGLFunctions_2_1>

--- a/Linear_cell_complex/examples/Linear_cell_complex/linear_cell_complex_3_viewer_qt.h
+++ b/Linear_cell_complex/examples/Linear_cell_complex/linear_cell_complex_3_viewer_qt.h
@@ -22,7 +22,7 @@
 
 #include <QApplication>
 #include <QKeyEvent>
-
+#include <QMap>
 #include <QGLViewer/qglviewer.h>
 #include <QKeyEvent>
 #include <QOpenGLFunctions_2_1>

--- a/Periodic_3_triangulation_3/demo/Periodic_3_triangulation_3/Viewer.h
+++ b/Periodic_3_triangulation_3/demo/Periodic_3_triangulation_3/Viewer.h
@@ -1,6 +1,6 @@
 #ifndef VIEWER_H
 #define VIEWER_H
-
+#include <QMap>
 #include <QGLViewer/qglviewer.h>
 
 class Viewer : public QGLViewer{

--- a/Periodic_3_triangulation_3/demo/Periodic_Lloyd_3/MainWindow.cpp
+++ b/Periodic_3_triangulation_3/demo/Periodic_Lloyd_3/MainWindow.cpp
@@ -150,7 +150,11 @@ MainWindow::savePoints()
 
 void MainWindow::lloydStep() {
   scene.lloyd_step();
+#if QGLVIEWER_VERSION >= 0x020700
+  viewer->update();
+#else
   viewer->updateGL();
+#endif
   viewer->changed();
   }
 

--- a/Periodic_3_triangulation_3/demo/Periodic_Lloyd_3/Viewer.h
+++ b/Periodic_3_triangulation_3/demo/Periodic_Lloyd_3/Viewer.h
@@ -2,6 +2,7 @@
 #define VIEWER_H
 
 #include "Scene.h"
+#include <QMap>
 #include <QGLViewer/qglviewer.h>
 #include <QOpenGLFunctions_2_1>
 #include <QOpenGLVertexArrayObject>

--- a/Polyhedron/demo/Polyhedron/MainWindow.cpp
+++ b/Polyhedron/demo/Polyhedron/MainWindow.cpp
@@ -1024,7 +1024,8 @@ void MainWindow::open(QString filename)
     default:
       load_pair = File_loader_dialog::getItem(fileinfo.fileName(), selected_items, &ok);
   }
-  viewer->context()->makeCurrent();
+
+  viewer->makeCurrent();
   if(!ok || load_pair.first.isEmpty()) { return; }
   
   if (load_pair.second)

--- a/Polyhedron/demo/Polyhedron/Plugins/Operations_on_polyhedra/Clip_polyhedron_plugin.cpp
+++ b/Polyhedron/demo/Polyhedron/Plugins/Operations_on_polyhedra/Clip_polyhedron_plugin.cpp
@@ -194,7 +194,7 @@ public Q_SLOTS:
                                               plane->plane(),
                                               ui_widget.close_checkBox->isChecked());
           poly->invalidateOpenGLBuffers();
-          viewer->updateGL();
+
         }
         else
         {
@@ -223,10 +223,9 @@ public Q_SLOTS:
           new_item->setVisible(poly->visible());
           scene->addItem(new_item);
           new_item->invalidateOpenGLBuffers();
-
-          viewer->updateGL();
         }
       }
+      viewer->update();
     }
     QApplication::restoreOverrideCursor();
   }

--- a/Polyhedron/demo/Polyhedron/Plugins/Point_set/Point_set_selection_plugin.cpp
+++ b/Polyhedron/demo/Polyhedron/Plugins/Point_set/Point_set_selection_plugin.cpp
@@ -252,8 +252,12 @@ protected:
       Qt::KeyboardModifiers modifiers = keyEvent->modifiers();
 
       shift_pressing = modifiers.testFlag(Qt::ShiftModifier);
-
+#if QGLVIEWER_VERSION >= 0x020700
+      background = static_cast<CGAL::Three::Viewer_interface*>(*QGLViewer::QGLViewerPool().begin())->grabFramebuffer();
+#else
       background = static_cast<CGAL::Three::Viewer_interface*>(*QGLViewer::QGLViewerPool().begin())->grabFrameBuffer();
+
+#endif
     }
 
     // mouse events

--- a/Polyhedron/demo/Polyhedron/Scene_polyhedron_item_k_ring_selection.h
+++ b/Polyhedron/demo/Polyhedron/Scene_polyhedron_item_k_ring_selection.h
@@ -567,7 +567,11 @@ protected:
       pen.setColor(QColor(Qt::green));
       pen.setWidth(3);
       //Create a QImage of the screen and paint the lasso on top of it
+#if QGLVIEWER_VERSION >= 0x020700
+      QImage image = viewer->grabFramebuffer();
+#else
       QImage image = viewer->grabFrameBuffer();
+#endif
       painter->begin(viewer);
       painter->drawImage(QPoint(0,0), image);
       painter->setPen(pen);

--- a/Principal_component_analysis/demo/Principal_component_analysis/MainWindow.cpp
+++ b/Principal_component_analysis/demo/Principal_component_analysis/MainWindow.cpp
@@ -181,7 +181,11 @@ void MainWindow::on_actionCopy_snapshot_triggered()
   QClipboard *qb = QApplication::clipboard();
   m_pViewer->makeCurrent();
   m_pViewer->raise();
+#if QGLVIEWER_VERSION >= 0x020700
+  QImage snapshot = m_pViewer->grabFramebuffer();
+#else
   QImage snapshot = m_pViewer->grabFrameBuffer(true);
+#endif
   qb->setImage(snapshot);
 	QApplication::restoreOverrideCursor();
 }

--- a/Principal_component_analysis/demo/Principal_component_analysis/Viewer.h
+++ b/Principal_component_analysis/demo/Principal_component_analysis/Viewer.h
@@ -1,6 +1,6 @@
 #ifndef VIEWER_H
 #define VIEWER_H
-
+#include <QMap>
 #include <QGLViewer/qglviewer.h>
 
 // forward declarations

--- a/Surface_mesher/demo/Surface_mesher/viewer.h
+++ b/Surface_mesher/demo/Surface_mesher/viewer.h
@@ -1,6 +1,7 @@
 #ifndef _VIEWER_H
 #define _VIEWER_H
 
+#include <QMap>
 #include <QGLViewer/qglviewer.h>
 
 class Surface;

--- a/Surface_mesher/demo/Surface_mesher/volume.cpp
+++ b/Surface_mesher/demo/Surface_mesher/volume.cpp
@@ -1243,7 +1243,9 @@ void Volume::draw()
   if(!m_view_mc && m_draw_triangulation)
   {
     // draw the triangualtion
-    mw->viewer->qglColor(m_triangulation_color);
+    ::glColor3d(m_triangulation_color.redF(),
+                m_triangulation_color.greenF(),
+                m_triangulation_color.blueF());
     ::glLineWidth(1.0);
     ::glBegin(GL_LINES);
     for(Tr::Finite_edges_iterator 
@@ -1362,7 +1364,9 @@ void Volume::gl_draw_surface()
 	} else {
 	  ::glNormal3d(n.x(),n.y(),n.z());
 	}
-	mw->viewer->qglColor(values_list->color(values_list->search(facet_cell->info())));
+        ::glColor3d(values_list->color(values_list->search(facet_cell->info())).redF(),
+                    values_list->color(values_list->search(facet_cell->info())).greenF(),
+                    values_list->color(values_list->search(facet_cell->info())).blueF());
 	::glVertex3d(a.x(),a.y(),a.z());
 	::glVertex3d(b.x(),b.y(),b.z());
 	::glVertex3d(c.x(),c.y(),c.z());
@@ -1378,7 +1382,9 @@ void Volume::gl_draw_surface()
     {
       if(values_list->enabled(i))
       {
-        mw->viewer->qglColor(values_list->color(i));
+        ::glColor3d(values_list->color(i).redF(),
+                  values_list->color(i).greenF(),
+                  values_list->color(i).blueF());
         ::glCallList(lists_draw_surface[i]);
       }
     }
@@ -1398,7 +1404,9 @@ void Volume::gl_draw_surface()
         % lists_draw_surface[i]
         % i;
         
-      mw->viewer->qglColor(values_list->color(i));
+      ::glColor3d(values_list->color(i).redF(),
+                values_list->color(i).greenF(),
+                values_list->color(i).blueF());
 
       if(!direct_draw && lists_draw_surface[i]) // If
         ::glNewList(lists_draw_surface[i],      // lists_draw_surface[i]==0

--- a/Three/include/CGAL/Three/Viewer_interface.h
+++ b/Three/include/CGAL/Three/Viewer_interface.h
@@ -23,7 +23,7 @@
 
 #include <CGAL/license/Three.h>
 
-
+#include <QMap>
 #include <QGLViewer/qglviewer.h>
 #include <QWidget>
 #include <QPoint>

--- a/Triangulation_3/demo/Triangulation_3/Scene.h
+++ b/Triangulation_3/demo/Triangulation_3/Scene.h
@@ -2,6 +2,7 @@
 #define SCENE_H
 
 #include "typedefs.h"
+#include <QMap>
 #include <QGLViewer/qglviewer.h>
 
 class Scene {

--- a/Triangulation_3/demo/Triangulation_3/Viewer.cpp
+++ b/Triangulation_3/demo/Triangulation_3/Viewer.cpp
@@ -1234,7 +1234,7 @@ void Viewer::draw()
         // Insert point mode
         if( m_curMode == INSERT_PT) {
             // Show prompt messages
-            qglColor( ::Qt::black );
+            ::glColor3d(0,0,0);
             drawText( width()-200, 20, tr("Shift+Left: Insert a point"), fontPrompt );
             drawText( width()-200, 40, tr("Hold Left: Move the point"), fontPrompt );
             drawText( width()-200, 60, tr("Return: Insert to DT"), fontPrompt );
@@ -1258,7 +1258,7 @@ void Viewer::draw()
         }
         else if( m_curMode == SELECT) {
             // Show prompt messages
-            qglColor( ::Qt::black );
+            ::glColor3d(0,0,0);
             drawText( width()-200, 20, tr("Shift+Left: Select"), fontPrompt );
             drawText( width()-200, 40, tr("Ctrl+Left: Add selection"),
                       QFont("Arial", 14) );
@@ -1274,7 +1274,7 @@ void Viewer::draw()
         }
         else if( m_curMode == MOVE ) {
             // Show prompt messages
-            qglColor( ::Qt::black );
+            ::glColor3d(0,0,0);
             drawText( width()-200, 20, tr("Left Click: Select"), fontPrompt );
             if(m_isMoving)
                 drawText( width()-200, 40, tr("Shift+Wheel: Resize trackball"), fontPrompt );
@@ -1288,7 +1288,7 @@ void Viewer::draw()
         }
         else if( m_curMode == FINDNB ) {
             // Show prompt messages
-            qglColor( ::Qt::black );
+            ::glColor3d(0,0,0);
             drawText( width()-200, 20, tr("Shift+Left: Place query point"), fontPrompt );
             drawText( width()-200, 40, tr("Shift+Wheel: Resize trackball"), fontPrompt );
             rendering_program.bind();
@@ -1307,7 +1307,7 @@ void Viewer::draw()
         }
         else if(m_curMode == EMPTYSPH){
             // Show prompt messages
-            qglColor( ::Qt::black );
+            ::glColor3d(0,0,0);
             drawText( width()-200, 20, tr("Shift+Left: Place query point"), fontPrompt );
             drawText( width()-200, 40, tr("Press S: Show/Hide trackball"), fontPrompt );
             drawText( width()-200, 60, tr("Shift+Wheel: Resize trackball"), fontPrompt );
@@ -1337,7 +1337,7 @@ void Viewer::draw()
             switch( m_curStep ) {
             case NEWPT:
                 // Show prompt messages
-                qglColor( ::Qt::black );
+                ::glColor3d(0,0,0);
                 drawText( 10, 20, tr("Highlight the next-to-insert point"), fontPrompt );
                 // Highlight the next-to-insert point
                 rendering_program.bind();
@@ -1351,7 +1351,7 @@ void Viewer::draw()
                 break;
             case CELL:  // show the tetrahedron that contains the point
                 // Show prompt messages
-                qglColor( ::Qt::black );
+                ::glColor3d(0,0,0);
                 drawText( 10, 20, tr("Show the tetrahedron containing the point"), fontPrompt );
                 drawText( 10, 40, tr("(Only finite facets are drawn)"), fontPrompt );
                 // Highlight the next-to-insert vertex
@@ -1373,7 +1373,7 @@ void Viewer::draw()
                 break;
             case CONFLICT:  // show the conflict region
                 // Show prompt messages
-                qglColor( ::Qt::black );
+                ::glColor3d(0,0,0);
                 drawText( 10, 20, tr("Show the conflict region"), fontPrompt );
                 // Highlight the next-to-insert vertex
                 rendering_program.bind();
@@ -1432,7 +1432,7 @@ void Viewer::draw()
 
         if( m_curMode == INSERT_PT) {
             // Show prompt messages
-            qglColor( ::Qt::black );
+            ::glColor3d(0,0,0);
             drawText( width()-200, 20, tr("Shift+Left: Insert a point"), fontPrompt );
             drawText( width()-200, 40, tr("Hold Left: Move the point"), fontPrompt );
             drawText( width()-200, 60, tr("Return: Insert to DT"), fontPrompt );
@@ -1457,7 +1457,7 @@ void Viewer::draw()
         }
         else if( m_curMode == SELECT) {
             // Show prompt messages
-            qglColor( ::Qt::black );
+            ::glColor3d(0,0,0);
             drawText( width()-200, 20, tr("Shift+Left: Select"), fontPrompt );
             drawText( width()-200, 40, tr("Ctrl+Left: Add selection"),
                       QFont("Arial", 14) );
@@ -1473,7 +1473,7 @@ void Viewer::draw()
         }
         else if( m_curMode == MOVE ) {
             // Show prompt messages
-            qglColor( ::Qt::black );
+            ::glColor3d(0,0,0);
             drawText( width()-200, 20, tr("Left Click: Select"), fontPrompt );
             if(m_isMoving)
                 drawText( width()-200, 40, tr("Shift+Wheel: Resize trackball"), fontPrompt );
@@ -1487,7 +1487,7 @@ void Viewer::draw()
         }
         else if( m_curMode == FINDNB ) {
             // Show prompt messages
-            qglColor( ::Qt::black );
+            ::glColor3d(0,0,0);
             drawText( width()-200, 20, tr("Shift+Left: Place query point"), fontPrompt );
             drawText( width()-200, 40, tr("Shift+Wheel: Resize trackball"), fontPrompt );
             rendering_program_spheres.bind();
@@ -1507,7 +1507,7 @@ void Viewer::draw()
         }
         else if(m_curMode == EMPTYSPH){
             // Show prompt messages
-            qglColor( ::Qt::black );
+            ::glColor3d(0,0,0);
             drawText( width()-200, 20, tr("Shift+Left: Place query point"), fontPrompt );
             drawText( width()-200, 40, tr("Press S: Show/Hide trackball"), fontPrompt );
             drawText( width()-200, 60, tr("Shift+Wheel: Resize trackball"), fontPrompt );
@@ -1539,7 +1539,7 @@ void Viewer::draw()
             switch( m_curStep ) {
             case NEWPT:
                 // Show prompt messages
-                qglColor( ::Qt::black );
+                ::glColor3d(0,0,0);
                 drawText( 10, 20, tr("Highlight the next-to-insert point"), fontPrompt );
                 // Highlight the next-to-insert point
                 rendering_program_spheres.bind();
@@ -1552,7 +1552,7 @@ void Viewer::draw()
                 break;
             case CELL:  // show the tetrahedron that contains the point
                 // Show prompt messages
-                qglColor( ::Qt::black );
+                ::glColor3d(0,0,0);
                 drawText( 10, 20, tr("Show the tetrahedron containing the point"), fontPrompt );
                 drawText( 10, 40, tr("(Only finite facets are drawn)"), fontPrompt );
                 // Highlight the next-to-insert vertex
@@ -1573,7 +1573,7 @@ void Viewer::draw()
                 break;
             case CONFLICT:  // show the conflict region
                 // Show prompt messages
-                qglColor( ::Qt::black );
+                ::glColor3d(0,0,0);
                 drawText( 10, 20, tr("Show the conflict region"), fontPrompt );
                 // Highlight the next-to-insert vertex
                 rendering_program_spheres.bind();
@@ -1610,7 +1610,7 @@ void Viewer::draw()
     }
     if( m_curMode == INSERT_V  ) {
         // Show prompt messages
-        qglColor( ::Qt::black );
+        ::glColor3d(0,0,0);
         drawText( width()-200, 20, tr("Shift+Left: Insert a vertex"), fontPrompt );
         drawText( width()-200, 40, tr("Shift+Wheel: Resize trackball"), fontPrompt );
 
@@ -1798,7 +1798,15 @@ void Viewer::mousePressEvent(QMouseEvent *event)
             select( event->pos() );
             if( m_isMoving )
                 // redraw window
-            {  changed(); updateGL();}
+            {  changed();
+
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
+            }
             else
                 // if no point is selected, then regular action (rotation) will be performed
                 QGLViewer::mousePressEvent(event);
@@ -1814,7 +1822,12 @@ void Viewer::mousePressEvent(QMouseEvent *event)
             // initialize multiple selection window
             m_rectSel = QRect( event->pos(), event->pos() );
             // redraw window
-            updateGL();
+#if QGLVIEWER_VERSION >= 0x020700
+            update();
+#else
+            updateGL()
+
+#endif
             break;
         case Qt::CTRL : // add selection
             m_isPress = true;
@@ -1822,7 +1835,12 @@ void Viewer::mousePressEvent(QMouseEvent *event)
             // initialize multiple selection window
             m_rectSel = QRect( event->pos(), event->pos() );
             // redraw window
-            updateGL();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
             break;
         default: // rotate
             QGLViewer::mousePressEvent(event);
@@ -1838,7 +1856,14 @@ void Viewer::mousePressEvent(QMouseEvent *event)
         // perform the selection
         select( event->pos() );
         if( m_isMoving ) // redraw window
-        {  changed(); updateGL();}
+        {  changed();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
+        }
         else // if no point is selected, then regular action (rotation) will be performed
             QGLViewer::mousePressEvent(event);
     }//end-if-move
@@ -1874,7 +1899,13 @@ void Viewer::mouseMoveEvent(QMouseEvent *event)
             // compute the conflict hole induced by point p
             computeConflict( m_newPt );
             // redraw
-            changed(); updateGL();
+            changed();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
         }//end-if-compute
     }//end-if-inspt
 
@@ -1926,7 +1957,13 @@ void Viewer::mouseReleaseEvent(QMouseEvent *event)
         }//end-if-compute
 
         // redraw
-        changed(); updateGL();
+        changed();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
     }//end-if-ins
 
     /* INS_PT mode - Shift+Left: compute and insert a point */
@@ -1941,7 +1978,13 @@ void Viewer::mouseReleaseEvent(QMouseEvent *event)
         }//end-if-compute
 
         // redraw
-        changed(); updateGL();
+        changed();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
     }//end-if-inspt
 
     /* INS_PT mode - Left: compute and insert a point */
@@ -1955,7 +1998,13 @@ void Viewer::mouseReleaseEvent(QMouseEvent *event)
         }//end-if-compute
 
         // redraw
-        changed(); updateGL();
+        changed();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
     }//end-if-inspt
 
     /* SEL mode - Left: terminate multiple point selection */
@@ -1997,7 +2046,13 @@ void Viewer::mouseReleaseEvent(QMouseEvent *event)
         }//end-if-selwindow
 
         // update display to show
-        changed(); updateGL();
+        changed();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
     }//end-if-select
 
     /* MOVE mode - Left: terminate point moving */
@@ -2024,7 +2079,14 @@ void Viewer::mouseReleaseEvent(QMouseEvent *event)
         }//end-if-compute
 
         // redraw
-        changed(); updateGL();
+        changed();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
+
     }//end-if-move
 
     /* FindNb mode - Shift+Left: find the nearest neighbor of the point */
@@ -2037,7 +2099,13 @@ void Viewer::mouseReleaseEvent(QMouseEvent *event)
         }//end-if-compute
 
         // redraw
-        changed(); updateGL();
+        changed();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
     }//end-if-findnb
 
     /* EmptySphere mode - Shift+Left: show the empty sphere of the cell */
@@ -2062,7 +2130,13 @@ void Viewer::mouseReleaseEvent(QMouseEvent *event)
             }
         }//end-if-compute
         // redraw
-        changed(); updateGL();
+        changed();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
     }//end-if-emptysphere
 
     else
@@ -2086,7 +2160,13 @@ void Viewer::wheelEvent(QWheelEvent *event)
             m_fRadius = 0.1f;
 
         // redraw
-        changed(); updateGL();
+        changed();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
     }//end-if-insv
 
     else if( m_curMode == INSERT_PT && modifiers == Qt::SHIFT ) {
@@ -2107,7 +2187,13 @@ void Viewer::wheelEvent(QWheelEvent *event)
         }//end-if-conflict
 
         // redraw
-        changed(); updateGL();
+        changed();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
     }//end-if-inspt
 
     // resize the trackball when moving a point
@@ -2137,7 +2223,13 @@ void Viewer::wheelEvent(QWheelEvent *event)
         m_pScene->m_vhArray[m_vidMoving] = vh;
 
         // redraw
-        changed(); updateGL();
+        changed();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
     }//end-if-move
 
     else
@@ -2173,7 +2265,13 @@ void Viewer::keyPressEvent(QKeyEvent *event)
         m_conflictCells.clear();
 
         // redraw
-        changed(); updateGL();
+        changed();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
     }//end-if-insVertex
 
     /* Cancel the newly inserted point and its conflict region */
@@ -2185,7 +2283,13 @@ void Viewer::keyPressEvent(QKeyEvent *event)
         m_conflictCells.clear();
 
         // redraw
-        changed(); updateGL();
+        changed();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
     }//end-if-escapeIns
 
     /* Delete selected points */
@@ -2202,7 +2306,13 @@ void Viewer::keyPressEvent(QKeyEvent *event)
         m_vidSeled.clear();
 
         // redraw
-        changed(); updateGL();
+        changed();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
     }//end-if-del
 
     /* Cancel the selection */
@@ -2215,7 +2325,13 @@ void Viewer::keyPressEvent(QKeyEvent *event)
         m_vidSeled.clear();
 
         // redraw
-        changed(); updateGL();
+        changed();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
     }//end-if-escapeSel
 
 
@@ -2224,7 +2340,13 @@ void Viewer::keyPressEvent(QKeyEvent *event)
              && event->key()==Qt::Key_S && modifiers==Qt::NoButton ) {
         m_showTrackball = !m_showTrackball;
         // redraw
-        changed(); updateGL();
+        changed();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
     }//end-if-showBall
 
     else
@@ -2232,7 +2354,13 @@ void Viewer::keyPressEvent(QKeyEvent *event)
 
 
     // redraw
-    changed(); updateGL();
+    changed();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
 }
 
 /*************************************************************/
@@ -2313,7 +2441,13 @@ void Viewer::toggleIncremental(bool on) {
     }
 
     // redraw
-    changed(); updateGL();
+    changed();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
 }
 
 void Viewer::stopIncremental() {
@@ -2337,7 +2471,13 @@ void Viewer::stopIncremental() {
     }
 
     // redraw
-    changed(); updateGL();
+    changed();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
 }
 
 void Viewer::incremental_insert() {
@@ -2396,7 +2536,13 @@ void Viewer::incremental_insert() {
     }
 
     // redraw
-    changed(); updateGL();
+    changed();
+#if QGLVIEWER_VERSION >= 0x020700
+              update();
+#else
+              updateGL()
+
+#endif
 }
 
 

--- a/Triangulation_3/demo/Triangulation_3/Viewer.cpp
+++ b/Triangulation_3/demo/Triangulation_3/Viewer.cpp
@@ -1803,7 +1803,7 @@ void Viewer::mousePressEvent(QMouseEvent *event)
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
             }
@@ -1825,7 +1825,7 @@ void Viewer::mousePressEvent(QMouseEvent *event)
 #if QGLVIEWER_VERSION >= 0x020700
             update();
 #else
-            updateGL()
+            updateGL();
 
 #endif
             break;
@@ -1838,7 +1838,7 @@ void Viewer::mousePressEvent(QMouseEvent *event)
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
             break;
@@ -1860,7 +1860,7 @@ void Viewer::mousePressEvent(QMouseEvent *event)
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
         }
@@ -1903,7 +1903,7 @@ void Viewer::mouseMoveEvent(QMouseEvent *event)
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
         }//end-if-compute
@@ -1913,7 +1913,7 @@ void Viewer::mouseMoveEvent(QMouseEvent *event)
         // update multiple selection window
         m_rectSel.setBottomRight( event->pos() );
         // redraw
-        //changed(); updateGL();
+        //changed(); updateGL();;
     }//end-if-sel
 
     else if( m_curMode == MOVE && m_isMoving ) {
@@ -1939,7 +1939,7 @@ void Viewer::mouseMoveEvent(QMouseEvent *event)
         }//end-if-compute
 
         // redraw
-        //  changed(); updateGL();
+        //  changed(); updateGL();;
     }//end-if-move
 
     else
@@ -1961,7 +1961,7 @@ void Viewer::mouseReleaseEvent(QMouseEvent *event)
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
     }//end-if-ins
@@ -1982,7 +1982,7 @@ void Viewer::mouseReleaseEvent(QMouseEvent *event)
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
     }//end-if-inspt
@@ -2002,7 +2002,7 @@ void Viewer::mouseReleaseEvent(QMouseEvent *event)
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
     }//end-if-inspt
@@ -2050,7 +2050,7 @@ void Viewer::mouseReleaseEvent(QMouseEvent *event)
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
     }//end-if-select
@@ -2083,7 +2083,7 @@ void Viewer::mouseReleaseEvent(QMouseEvent *event)
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
 
@@ -2103,7 +2103,7 @@ void Viewer::mouseReleaseEvent(QMouseEvent *event)
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
     }//end-if-findnb
@@ -2134,7 +2134,7 @@ void Viewer::mouseReleaseEvent(QMouseEvent *event)
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
     }//end-if-emptysphere
@@ -2164,7 +2164,7 @@ void Viewer::wheelEvent(QWheelEvent *event)
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
     }//end-if-insv
@@ -2191,7 +2191,7 @@ void Viewer::wheelEvent(QWheelEvent *event)
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
     }//end-if-inspt
@@ -2227,7 +2227,7 @@ void Viewer::wheelEvent(QWheelEvent *event)
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
     }//end-if-move
@@ -2269,7 +2269,7 @@ void Viewer::keyPressEvent(QKeyEvent *event)
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
     }//end-if-insVertex
@@ -2287,7 +2287,7 @@ void Viewer::keyPressEvent(QKeyEvent *event)
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
     }//end-if-escapeIns
@@ -2310,7 +2310,7 @@ void Viewer::keyPressEvent(QKeyEvent *event)
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
     }//end-if-del
@@ -2329,7 +2329,7 @@ void Viewer::keyPressEvent(QKeyEvent *event)
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
     }//end-if-escapeSel
@@ -2344,7 +2344,7 @@ void Viewer::keyPressEvent(QKeyEvent *event)
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
     }//end-if-showBall
@@ -2358,7 +2358,7 @@ void Viewer::keyPressEvent(QKeyEvent *event)
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
 }
@@ -2445,7 +2445,7 @@ void Viewer::toggleIncremental(bool on) {
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
 }
@@ -2475,7 +2475,7 @@ void Viewer::stopIncremental() {
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
 }
@@ -2540,7 +2540,7 @@ void Viewer::incremental_insert() {
 #if QGLVIEWER_VERSION >= 0x020700
               update();
 #else
-              updateGL()
+              updateGL();
 
 #endif
 }

--- a/Triangulation_3/demo/Triangulation_3/Viewer.h
+++ b/Triangulation_3/demo/Triangulation_3/Viewer.h
@@ -100,7 +100,13 @@ public:
     m_isMoving = false;
     m_hasEmptyS = false;
     m_nearestNb = NULL;
-    updateGL();
+#if QGLVIEWER_VERSION >= 0x020700
+    update();
+#else
+    updateGL()
+
+#endif
+
   }
 
   // set selectBuffer size (if necessary)
@@ -194,12 +200,55 @@ public Q_SLOTS:
   void incremental_insert();
 
   // show options
-  inline void toggleShowAxis(bool flag)  { m_showAxis = flag; updateGL(); }
-  inline void toggleShowVertex(bool flag)  { m_showVertex = flag; updateGL(); }
-  inline void toggleShowDEdge(bool flag)  { m_showDEdge = flag; updateGL(); }
-  inline void toggleShowVEdge(bool flag)  { m_showVEdge = flag; updateGL(); }
-  inline void toggleShowFacet(bool flag)  { m_showFacet = flag; updateGL(); }
-  inline void toggleFlat(bool flag)  { m_isFlat = flag; updateGL(); }
+  inline void toggleShowAxis(bool flag)  {
+    m_showAxis = flag;
+#if QGLVIEWER_VERSION >= 0x020700
+    update();
+#else
+    updateGL()
+
+#endif
+  }
+  inline void toggleShowVertex(bool flag)  { m_showVertex = flag;
+                                         #if QGLVIEWER_VERSION >= 0x020700
+                                             update();
+                                         #else
+                                             updateGL()
+
+                                         #endif
+                                           }
+  inline void toggleShowDEdge(bool flag)  { m_showDEdge = flag;
+                                        #if QGLVIEWER_VERSION >= 0x020700
+                                            update();
+                                        #else
+                                            updateGL()
+
+                                        #endif
+                                          }
+  inline void toggleShowVEdge(bool flag)  { m_showVEdge = flag;
+                                        #if QGLVIEWER_VERSION >= 0x020700
+                                            update();
+                                        #else
+                                            updateGL()
+
+                                        #endif
+                                          }
+  inline void toggleShowFacet(bool flag)  { m_showFacet = flag;
+                                        #if QGLVIEWER_VERSION >= 0x020700
+                                            update();
+                                        #else
+                                            updateGL()
+
+                                        #endif
+                                          }
+  inline void toggleFlat(bool flag)  { m_isFlat = flag;
+                                   #if QGLVIEWER_VERSION >= 0x020700
+                                       update();
+                                   #else
+                                       updateGL()
+
+                                   #endif
+                                     }
 
   // set preferences
   void setPreferences() {
@@ -226,7 +275,12 @@ public Q_SLOTS:
     m_iStep = m_pDlgPrefer->m_iStep*40;
     m_colorEmptySphere = m_pDlgPrefer->m_colorEmptySphere;
     // redraw
-    updateGL();
+#if QGLVIEWER_VERSION >= 0x020700
+    update();
+#else
+    updateGL()
+
+#endif
   }
 
   Q_SIGNALS:

--- a/Triangulation_3/demo/Triangulation_3/Viewer.h
+++ b/Triangulation_3/demo/Triangulation_3/Viewer.h
@@ -103,7 +103,7 @@ public:
 #if QGLVIEWER_VERSION >= 0x020700
     update();
 #else
-    updateGL()
+    updateGL();
 
 #endif
 
@@ -205,7 +205,7 @@ public Q_SLOTS:
 #if QGLVIEWER_VERSION >= 0x020700
     update();
 #else
-    updateGL()
+    updateGL();
 
 #endif
   }
@@ -213,7 +213,7 @@ public Q_SLOTS:
                                          #if QGLVIEWER_VERSION >= 0x020700
                                              update();
                                          #else
-                                             updateGL()
+                                             updateGL();
 
                                          #endif
                                            }
@@ -221,7 +221,7 @@ public Q_SLOTS:
                                         #if QGLVIEWER_VERSION >= 0x020700
                                             update();
                                         #else
-                                            updateGL()
+                                            updateGL();
 
                                         #endif
                                           }
@@ -229,7 +229,7 @@ public Q_SLOTS:
                                         #if QGLVIEWER_VERSION >= 0x020700
                                             update();
                                         #else
-                                            updateGL()
+                                            updateGL();
 
                                         #endif
                                           }
@@ -237,7 +237,7 @@ public Q_SLOTS:
                                         #if QGLVIEWER_VERSION >= 0x020700
                                             update();
                                         #else
-                                            updateGL()
+                                            updateGL();
 
                                         #endif
                                           }
@@ -245,7 +245,7 @@ public Q_SLOTS:
                                    #if QGLVIEWER_VERSION >= 0x020700
                                        update();
                                    #else
-                                       updateGL()
+                                       updateGL();
 
                                    #endif
                                      }
@@ -278,7 +278,7 @@ public Q_SLOTS:
 #if QGLVIEWER_VERSION >= 0x020700
     update();
 #else
-    updateGL()
+    updateGL();
 
 #endif
   }

--- a/Triangulation_3/demo/Triangulation_3/Viewer.h
+++ b/Triangulation_3/demo/Triangulation_3/Viewer.h
@@ -2,6 +2,7 @@
 #define VIEWER_H
 
 #include "Scene.h"
+#include <QMap>
 #include <QGLViewer/qglviewer.h>
 #include <QMouseEvent>
 #include <QKeyEvent>

